### PR TITLE
Pedro/aeq update

### PR DESCRIPTION
**Pull request by pedrocamargo**

### CODE REVIEW

**Comment by pedrocamargo** [comment]
Just a quick update and setting up sponsorship for the project

**Comment by pedrocamargo** [comment]
Anyone?

File: gis/desire_lines_procedure.py [hunk 1]
```diff
@@ -106,11 +106,13 @@ def doWork(self):
 
             if self.dl_type == "DesireLines":
                 items = []
+                max_zone = self.matrix.index[:].max().astype(np.int64)
                 for i, j in all_centroids.items():
```
Comment (jamiecook) [inline gis/desire_lines_procedure.py]: LIST COMPREHENSION!!!

```python
items = [(i,j[0],j[1]) for i, j in all_centroids.items() if i <= max_zone]
```

File: gis/desire_lines_procedure.py [hunk 2]
```diff
@@ -106,11 +106,13 @@ def doWork(self):
...
-                    items.append((i, j[0], j[1]))
+                    if i <= max_zone:
+                        items.append((i, j[0], j[1]))
                 coords = np.array(items)
 
-                coord_index = np.zeros((self.matrix.index[:].max().astype(np.int64) + 1, 2))
+                coord_index = np.zeros((max_zone + 1, 2))
```
Comment (jamiecook) [inline gis/desire_lines_procedure.py]: Also.. is this meant to be handling the case where you non-contiguous zones? if so - the following probably doesn't work right?

Comment (pedrocamargo) [reply gis/desire_lines_procedure.py]: No.. It is just that max_zone is the ID of the zone, but everything is zero_based

File: gis/desire_lines_procedure.py (hunk 1 above)
Comment (pedrocamargo) [reply gis/desire_lines_procedure.py]: Done!

File: gis/desire_lines_procedure.py (hunk 2 above)
Comment (pedrocamargo) [reply gis/desire_lines_procedure.py]: I think it is a little more clear now

**Comment by pedrocamargo** [comment]
Can you approve it, @jamiecook  ?

File: paths_procedures/show_shortest_path_dialog.py [hunk 3]
```diff
@@ -142,47 +141,41 @@ def produces_path(self):
...
-                # it = self.line_layer.getFeatures(QgsFeatureRequest(expr))
-                #
-                # ids = [i.id() for i in it]
-                # self.line_layer.setSelectedFeatures(ids)
-
+                # If you want to do selections instead of new layers
+                if self.rdo_selection.isChecked():
+                    f = self.cb_link_id_field
+                    t = ''
+                    for k in self.res.path[:-1]:
```
Comment (jamiecook) [inline paths_procedures/show_shortest_path_dialog.py:148]:
``` 
t = " or ".join([f"{f}={k}" for k in self.res.path])
```

File: paths_procedures/show_shortest_path_dialog.py [hunk 4]
```diff
@@ -142,47 +141,41 @@ def produces_path(self):
...
-                # add all links to the temp layer
-                pr.addFeatures(all_links)
-
-                # add layer to the map
-                QgsProject.instance().addMapLayer(vl)
-
-                symbol = vl.renderer().symbol()
-                symbol.setWidth(1)
-                qgis.utils.iface.mapCanvas().refresh()
+                else:
```
Comment (jamiecook) [inline paths_procedures/show_shortest_path_dialog.py:154]: have you considered having methods?

```
if (...):
    build_select_by_epxr()
else:
  create_new_layers()

File: paths_procedures/traffic_assignment_dialog.py [hunk 5]
```diff
@@ -574,7 +574,7 @@ def change_graph_settings(self):
                     skims.append(only_str(table.item(i, 1).text()))
 
         if len(skims) == 0:
-            skims = False
+            skims = []
```
Comment (jamiecook) [inline paths_procedures/traffic_assignment_dialog.py:577]: que? 

There is only one list that has len(x) == 0.... the empty list.... so i'm reading

```
if (skims == []): 
    skims = []

**Comment by jamiecook** [comment]
Did you make a bunch of black changes in this branch too?

I've been advocating for keeping those types of changes in their own PR

**Review by jamiecook — APPROVED** [review]

**Comment by pedrocamargo** [comment]
There is still some left over black changes here. Thy shouldn't come in future PRs, though.

File: paths_procedures/show_shortest_path_dialog.py (hunk 3 above)
Comment (pedrocamargo) [reply paths_procedures/show_shortest_path_dialog.py:148]: Added the change for release 0.6

File: paths_procedures/show_shortest_path_dialog.py (hunk 4 above)
Comment (pedrocamargo) [reply paths_procedures/show_shortest_path_dialog.py:154]: Indeed it is more elegant. Also added to version 0.6

**Pull request merged**